src/bar.jl

```diff
@@ -1,10 +1,22 @@
 @userplot GroupedBar
+GroupedBar(args...) = GroupedBar(args)
 
 grouped_xy(x::AbstractVector, y::AbstractMatrix) = x, y
 grouped_xy(y::AbstractMatrix) = 1:size(y,1), y
+grouped_xy(y::AbstractVector) = 1:length(y), reshape(y, length(y), 1)
+
+# handle grouping by DataFrame column
+function Plots.extractGroupArgs(group::Symbol, g::GroupedBar, args...)
+    Plots.extractGroupArgs(collect(g.args[1][group]))
+end
 
 @recipe function f(g::GroupedBar)
+    if isa(g.args[1], DataFrame)
+        d, g.args = processDF(d, g.args[1], g.args[2:end]...)
+    end
     x, y = grouped_xy(g.args...)
+    @show x
+    @show y
```

Comment:
Yes, this PR currently stinks. The WIP label doesn't lie. I was going to revamp the groupedbar recipe to accept a DataFrame based loosely on the suggestion in #44 but haven't made progress for months.